### PR TITLE
Adjust Fetcher after latest Experiments introduction

### PR DESCRIPTION
**Pull request by AleksanderWWW**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [UNRELEASED] neptune-fetcher 0.5.0
+
+## Breaking Changes
+- Dropped support for `sys/name` field and `names_regex` parameter in `ReadOnlyProject` ([#30](https://github.com/neptune-ai/neptune-fetcher/pull/30))
```
Comment (AleksanderWWW) [inline CHANGELOG.md:4]: To be honest, I'm not sure how to handle a case where we have a breaking change in the version <1.0. Will appreciate any guidance here.

**Comment by Raalsky** [comment]
Was you able to run our Functional tests?

File: README.md [hunk 2]
```diff
@@ -115,7 +115,7 @@ df = project.fetch_runs()
 ```
 
 __Returns__:
-`pandas.DataFrame` with three columns (`sys/id`, `sys/name` and `sys/custom_run_id`) and rows corresponding to project runs.
+`pandas.DataFrame` with two columns (`sys/id` and `sys/custom_run_id`) and rows corresponding to project runs.
```
Comment (Raalsky) [inline README.md:118]: `list_runs()` have a mention of names as well

File: README.md [hunk 3]
```diff
@@ -115,7 +115,7 @@ df = project.fetch_runs()
 ```
 
 __Returns__:
-`pandas.DataFrame` with three columns (`sys/id`, `sys/name` and `sys/custom_run_id`) and rows corresponding to project runs.
```
Comment (Raalsky) [inline README.md:118]: `fetch_runs()` have a mention of names in the method description.

File: README.md [hunk 4]
```diff
@@ -146,12 +146,12 @@ __Example__:
 ```python
 # Fetch all runs with specific columns
 runs_df = project.fetch_runs_df(
-    columns=["sys/name", "sys/modification_time", "training/lr"]
+    columns=["sys/custom_run_id", "sys/modification_time", "training/lr"]
```
Comment (Raalsky) [inline README.md:149]: Remove the `names_regex` parameter

File: README.md [hunk 5]
```diff
@@ -146,12 +146,12 @@ __Example__:
...
 )
 
 # Fetch all runs with specific columns and extra columns that match a regex pattern
 runs_df = project.fetch_runs_df(
-    columns=["sys/name", "sys/modification_time"],
+    columns=["sys/custom_run_id", "sys/modification_time"],
```
Comment (Raalsky) [inline README.md:154]: There is an example of how to use `names_regex` -> to remove

File: tests/features/steps/fetching_runs_steps.py [hunk 6]
```diff
@@ -60,7 +60,6 @@ def step_impl(context):
 @then("we should get second run")
 def step_impl(context):
     assert len(context.dataframe) == 1
-    assert context.dataframe["sys/name"].values[0] == f"{context.project_key}-2"
```
Comment (Raalsky) [inline tests/features/steps/fetching_runs_steps.py:63]: Doesn't match the step name. How about using the `custom_run_id` instead?

File: tests/features/steps/run_steps.py [hunk 7]
```diff
@@ -82,7 +82,6 @@ def step_impl(context):
             "sys/description",
             "sys/family",
             "sys/id",
-            "sys/name",
```
Comment (Raalsky) [inline tests/features/steps/run_steps.py:85]: This still be here. It's a proper field but we don't want to expose it in Fetcher API for Runs.

File: tests/integration/test_read_only_run.py [hunk 8]
```diff
@@ -53,7 +53,6 @@ def test__field_names(api_token, hosted_backend):
             "sys/failed",
             "sys/id",
             "sys/custom_run_id",
-            "sys/name",
```
Comment (Raalsky) [inline tests/integration/test_read_only_run.py:56]: Will still be here. Should be as the metadata itself exists, we just don't want to make it a part of our API like: explicit parameters to support names, a part of response dictionary etc.

File: tests/integration/conftest.py [hunk 9]
```diff
@@ -65,8 +65,8 @@ def create_leaderboard_entry(sys_id, custom_run_id, name, columns=None):
                 [
                     StringField(path="sys/id", value=sys_id),
                     StringField(path="sys/custom_run_id", value=custom_run_id),
-                    StringField(path="sys/name", value=name),
                     BoolField(path="sys/failed", value=True),
+                    StringField(path="sys/name", value=name),
```
Comment (Raalsky) [inline tests/integration/conftest.py:69]: Maybe we should use empty string here?

File: CHANGELOG.md (hunk 1 above)
Comment (Raalsky) [reply CHANGELOG.md:4]: Looks good. We should probably mention that we're only listing Runs without Experiments as that's a part of Backward Incompatibility.

**Review by Raalsky — APPROVED** [review]

**Comment by AleksanderWWW** [comment]
@normandy7 we're good engineering-wise here, could you take a look at the docs? :wink:

**Review by normandy7 — APPROVED** [review]
Improved the docs (clarified wording and added examples) in this commit: https://github.com/neptune-ai/neptune-fetcher/pull/30/commits/af628256941bb0aeb0a7b7d98554622e455ca214

Let me know if something's off. Otherwise, looks good.

**Pull request merged**